src/legacy/core_plugins/markdown_vis_type/index.ts

```diff
@@ -18,19 +18,19 @@
  */
 
 import { resolve } from 'path';
+import { Legacy } from '../../../../kibana';
+import { LegacyPluginApi } from '../../plugin_discovery/types';
 
-export default function (kibana) {
-
-  return new kibana.Plugin({
-
+// eslint-disable-next-line import/no-default-export
+export default function MarkdownVisTypePlugin(kibana: LegacyPluginApi) {
+  const config: Legacy.PluginSpecOptions = {
+    id: 'markdown_vis_type',
+    require: ['data', 'visualizations'],
     uiExports: {
-      visTypes: [
-        'plugins/markdown_vis/markdown_vis'
-      ],
-      interpreter: ['plugins/markdown_vis/markdown_fn'],
       styleSheetPaths: resolve(__dirname, 'public/index.scss'),
-    }
-
-  });
+      visTypes: ['plugins/markdown_vis_type/index'],
```

Comment:
i tried with public dir but that doesn't seem to work ? (what data plugin does for example) seems in that case we need to have something consuming the plugin.

Reply:
I asked few people last week what we should use to load shims, and create [a new `uiExport` in case we want to use that](https://github.com/elastic/kibana/pull/39360/files).

@joshdover suggested to simply use `hacks` `uiExport`.

@elastic/kibana-platform maybe it is worth putting into migration guide the suggested way to load client-side shims?

> i tried with public dir but that doesn't seem to work ?

I don't know what `publicDir` does, but my guess always was that it tells Webpack where is the "`./public`" folder of the plugin.